include/arch/x86/page.h

```diff
@@ -158,6 +158,12 @@ typedef unsigned long mfn_t;
 #define VA_BITS        32
 #endif
 
+#ifdef __i386__
+#define PT_LEVELS 3
```

Comment:
May deserve a comment that this implies PAE:
```c
#define PT_LEVELS 3     /* assumes PAE */
```

Reply:
I would leave it as is, since PAE means 4 or 3 levels (depending on PSE).

Reply:
Uhm, PSE only toggles if huge mappings (4MiB with CR4.PAE=0, 2MiB with CR4.PAE=1) are supported, not how many levels the page table walk has. PSE, however, may short the final lookup by one level, as the page table lookup no longer happens for page directory entries with the PS bit set.

For "regular" 32-bit 4k page lookups it's as follows:
- "legacy" paging uses two levels: page directory (1024 entries) -> page table (1024 entries) -> page (4KiB)
- PAE uses three: page-directory-pointer table (4 entries) -> page directory (512 entries) -> page table (512 entries) -> page (4KiB)

So defining `PT_LEVELS` to 3 really only makes sense when CR4.PAE is 1, supporting the rationale of adding a related comment.

Reply:
Ugh, of course you're right. I was thinking about something else (incorrectly).